src/vdaf/prio3.rs

```diff
@@ -1211,19 +1262,21 @@ where
             )));
         }
 
-        // Check the proof verifier.
-        match self.typ.decide(&verifier) {
-            Ok(true) => (),
-            Ok(false) => {
-                return Err(VdafError::Uncategorized(
-                    "proof verifier check failed".into(),
-                ))
+        // Check the proof verifiers.
+        for verifier in verifiers.chunks(self.typ.verifier_len()) {
+            match self.typ.decide(verifier) {
+                Ok(true) => (),
+                Ok(false) => {
+                    return Err(VdafError::Uncategorized(
+                        "proof verifier check failed".into(),
+                    ))
+                }
+                Err(err) => return Err(VdafError::from(err)),
             }
```

Comment:
```suggestion
            if !self.typ.decide(verifier)? {
                return Err(VdafError::Uncategorized(
                    "proof verifier check failed".into(),
                ));
            }
```
nit: looks like we derive `FlpError` in `VdafError`, so we can use the `?` operator to clean this up a bit.

Reply:
Nice!